tests/test_task.py

```diff
@@ -106,66 +101,122 @@ def test_post_init_derived_attributes(
         task.session["v"] - task.catch_prob * task.session["v"],
         atol=0.1,
     )  # within 10% tolerance
     assert np.isclose(
-        counts["a"] / len(modality_seq),
+        counts["a"] / len(task._modality_seq),
         task.session["a"] - task.catch_prob * task.session["a"],
         atol=0.1,
     )
     assert np.isclose(
-        (counts["a"] + counts["v"] + counts["catch"]) / len(modality_seq),
+        (counts["a"] + counts["v"] + counts["catch"]) / len(task._modality_seq),
         1,
-        atol=0.05,
+        atol=0.1,
     )
 
 
 def test_build_trials_seq_shuffling():
     task_shuffled = Task(NAME, shuffle_trials=True)
     task_not_shuffled = Task(NAME, shuffle_trials=False)
 
-    task_shuffled._ntrials = NTRIALS
-    task_not_shuffled._ntrials = NTRIALS
-
-    task_shuffled._rng = np.random.default_rng(NTRIALS)
-    task_not_shuffled._rng = np.random.default_rng(NTRIALS)
-
-    sequence_shuffled = task_shuffled._build_trials_seq()
-    sequence_not_shuffled = task_not_shuffled._build_trials_seq()
+    _ = task_shuffled.generate_trials(ntrials=NTRIALS)
+    _ = task_not_shuffled.generate_trials(ntrials=NTRIALS)
 
     # Verify that the generated sequences are shuffled or not shuffled accordingly
-    assert sequence_shuffled.shape == sequence_not_shuffled.shape
-    assert not np.array_equal(sequence_shuffled, sequence_not_shuffled)
+    assert task_shuffled._modality_seq.shape == task_not_shuffled._modality_seq.shape
+    assert not np.array_equal(task_shuffled._modality_seq, task_not_shuffled._modality_seq)
 
 
 def test_build_trials_seq_maximum_sequential_trials():
     # Create a Task instance with shuffling enabled and a maximum sequential trial constraint
     task = Task(name=NAME, max_sequential=4)
-    task._ntrials = NTRIALS
-    task._rng = np.random.default_rng(NTRIALS)
-    modality_seq = task._build_trials_seq()
+    _ = task.generate_trials(ntrials=NTRIALS)
     # Ensure that no more than the specified maximum number of consecutive trials of the same modality occur
     for modality in task._modalities:
-        for i in range(len(modality_seq) - task.max_sequential):
-            assert np.sum(modality_seq[i : i + task.max_sequential] == modality) <= task.max_sequential
+        for i in range(len(task._modality_seq) - task.max_sequential):
+            assert np.sum(task._modality_seq[i : i + task.max_sequential] == modality) <= task.max_sequential
+
+
+@pytest.mark.parametrize(
+    ("stim_time", "fix_time", "iti"),
+    [(1000, 100, 0), (1000, 100, (300, 500))],
+)
+def test_setup_trial_phases(stim_time: int, fix_time: int, iti: int | tuple[int, int]):
+    task = Task(NAME, stim_time=stim_time, fix_time=fix_time, iti=iti)
+    _ = task.generate_trials(ntrials=NTRIALS)
+    trial_indices = range(NTRIALS)
+    # iti
+    assert task._iti.shape == (NTRIALS,)
+    if type(iti) is tuple:
+        assert min(task._iti) == min(iti)
+        assert max(task._iti) == max(iti)
```

Comment:
Is this always true? Doesn't it depend on the random number generator whether a min and a max is drawn?

Also, should we give a default random seed to the test task instance to ensure that the tests are reproducible?

```suggestion
        assert min(task._iti) >= min(iti)
        assert max(task._iti) <= max(iti)
```

Reply:
Good catch, I edited the comparisons. I'm against fixing a seed since this should always work. In general, if we fix a seed we may not catch some errors.